src/compression.rs

```diff
@@ -187,6 +187,7 @@ pub struct Pattern {
 }
 
 impl Expr {
+    #[allow(clippy::ptr_arg)]
```

Comment:
`clippy` wants `zip: &[ZNode]`, because that's a more general type than `Zip == Vec<ZNode>`. However, I think this is less legible in this instance, so I've just disabled the warning.